modules/c8000v/main.tf

```diff
@@ -0,0 +1,66 @@
+data "equinix_network_device_type" "this" {
+  category = "ROUTER"
+  vendor   = "Cisco"
+}
+
+data "equinix_network_device_platform" "this" {
+  device_type = data.equinix_network_device_type.this.code
+  flavor      = var.platform
+}
+
+data "equinix_network_device_software" "this" {
+  device_type = data.equinix_network_device_type.this.code
+  packages    = [var.software_package]
+  stable      = true
+  most_recent = true
+}
+
+resource "equinix_network_device" "non_cluster" {
+  lifecycle {
+    ignore_changes = [version, core_count]
+    precondition {
+      condition     = length(var.hostname) >= 2 && length(var.hostname) <= 10
+      error_message = "Device hostname should consist of 2 to 10 characters."
+    }
+  }
+  self_managed         = true
+  byol                 = var.byol
+  name                 = var.name
+  project_id           = var.project_id
+  hostname             = var.hostname
+  type_code            = data.equinix_network_device_type.this.code
+  package_code         = var.software_package
+  version              = data.equinix_network_device_software.this.version
+  core_count           = data.equinix_network_device_platform.this.core_count
+  metro_code           = var.metro_code
+  connectivity         = var.connectivity
+  account_number       = var.account_number
+  license_token        = var.byol ? var.license_token : null
+  term_length          = var.term_length
+  interface_count      = var.interface_count
+  notifications        = var.notifications
+  acl_template_id      = var.acl_template_id != "" ? var.acl_template_id : null
+  additional_bandwidth = var.additional_bandwidth > 0 ? var.additional_bandwidth : null
```

Comment:
The default value should be 0, not null.

Reply:
updated

Reply:
as discussed its going to be as is